### PR TITLE
VPC E-W security policy resource

### PR DESCRIPTION
**Pull request by ksamoray**
Implement the resource for east-west VPC security policy.

### CODE REVIEW

File: nsxt/provider.go [hunk 1]
```diff
@@ -1201,28 +1202,35 @@ func getGlobalPolicyEnforcementPointPath(m interface{}, sitePath *string) string
...
 
-func getProjectIDFromSchema(d *schema.ResourceData) string {
+func getContextDataFromSchema(d *schema.ResourceData) (string, string) {
 	ctxPtr := d.Get("context")
 	if ctxPtr != nil {
 		contexts := ctxPtr.([]interface{})
 		for _, context := range contexts {
 			data := context.(map[string]interface{})
+			vpcID := ""
+			if data["vpi_id"] != nil {
```
[the remark this answers is not in the document]
Comment (ksamoray) [reply nsxt/provider.go:1212]: Indeed

**Comment by annakhm** [comment]
This PR does not cover the provider-defined VPC scope yet, correct?

**Comment by ksamoray** [comment]
> This PR does not cover the provider-defined VPC scope yet, correct?

Indeed - the provider-level VPC specification is a global, non per-resource change.

File: nsxt/provider.go [hunk 2]
```diff
@@ -496,6 +496,7 @@ func Provider() *schema.Provider {
 			"nsxt_policy_gateway_flood_protection_profile_binding":     resourceNsxtPolicyGatewayFloodProtectionProfileBinding(),
 			"nsxt_policy_compute_sub_cluster":                          resourceNsxtPolicyComputeSubCluster(),
 			"nsxt_policy_tier0_inter_vrf_routing":                      resourceNsxtPolicyTier0InterVRFRouting(),
+			"nsxt_policy_vpc_security_policy":                          resourceNsxtPolicyVPCSecurityPolicy(),
```
Comment (annakhm) [inline nsxt/provider.go:499]: can we remove the first `policy` for all VPC resources? There is no other VPC.. Would love to hear also from @salv-orlando and @tvigneron

**Comment by qiyueyao** [comment]
Hi, a quick maybe dumb question: does this series of change support `data_source_nsxt_policy_security_policy` and `data_source_nsxt_policy_gateway_policy`? Looks like for Get with ID, it's handled, but without ID, it still lists the `domain` 🤔

**Comment by ksamoray** [comment]
> Hi, a quick maybe dumb question: does this series of change support `data_source_nsxt_policy_security_policy` and `data_source_nsxt_policy_gateway_policy`? Looks like for Get with ID, it's handled, but without ID, it still lists the `domain` 🤔

@qiyueyao I haven't got to handle these data sources yet but yeah - unless they use search API, they'll be affected by this change as the API wrapper is modified.
I believe that I should add a check to the DS, that the VPC is not specified - as these datasources aren't supposed to serve VPC.
VPC should have a pair of `data_source_nsxt_policy_vpc.*` datasources.

File: nsxt/provider.go (hunk 2 above)
Comment (salv-orlando) [reply nsxt/provider.go:499]: I agree, also proposed that in design review.

Comment (tvigneron) [reply nsxt/provider.go:499]: It seems a great idea - for vpc resources let's get rid of the Policy. We don't want the name to be unecessarily long

File: nsxt/resource_nsxt_vpc_security_policy_test.go [hunk 3]
```diff
@@ -0,0 +1,174 @@
...
+	tag2 := "def"
+
+	resource.ParallelTest(t, resource.TestCase{
+		PreCheck: func() {
+			testAccPreCheck(t)
+			testAccOnlyVPC(t)
+		},
+		Providers: testAccProviders,
+		CheckDestroy: func(state *terraform.State) error {
+			return testAccNsxtPolicySecurityPolicyCheckDestroy(state, updatedName, defaultDomain)
```
Comment (ksamoray) [inline nsxt/resource_nsxt_vpc_security_policy_test.go:36]: Should pass the resource name or use a VPC func here

File: nsxt/resource_nsxt_policy_predefined_security_policy.go [hunk 4]
```diff
@@ -409,20 +464,29 @@ func resourceNsxtPolicyPredefinedSecurityPolicyDelete(d *schema.ResourceData, m
...
+		childVPC, err := createChildVPCWithSecurityPolicy(context, *policy.Id, policy)
+		if err != nil {
+			return fmt.Errorf("Failed to create H-API for VPC Security Policy: %s", err)
+		}
+		orgRoot := model.OrgRoot{
+			ResourceType: strPtr("OrgRoot"),
+			Children:     []*data.StructValue{childVPC},
+		}
+
+		client := nsxt.NewOrgRootClient(connector)
```
Comment (annakhm) [inline nsxt/resource_nsxt_policy_predefined_security_policy.go:478]: This would work with VPC admin priviledge?

**Review by annakhm — APPROVED** [review]

File: website/docs/r/vpc_security_policy.html.markdown [hunk 5]
```diff
@@ -0,0 +1,131 @@
...
+
+An existing security policy can be [imported][docs-import] into this resource, via the following command:
+
+[docs-import]: https://www.terraform.io/cli/import
+
+```
+terraform import nsxt_vpc_security_policy.policy1 PATH
+```
+
+The above command imports the VPC security policy named `policy1` under NSX domain `domain` with the NSX Policy path `PATH`.
```
Comment (annakhm) [inline website/docs/r/vpc_security_policy.html.markdown:131]: `domain` is irrelevant here

**Comment by annakhm** [comment]
LGTM. There are some doc fixes needed, but feel free to address them in one of the following PRs if you prefer

**Pull request merged**